users/UserRouter.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express');
+const router = express.Router();
+const session = require('express-session');
+
+const User = require('./userModel.js');
+
+//middleware for protect when get users.  user needs to login before get users.  However, this middleware is local used in GET users route only.
+function protected(req, res, next) {
+    if(req.session && req.session.username) {
+        next();
+    } else {
+        res.status(401).json({ message: 'You shall NOT pass!'})
+    }
+}
+
+//start endpoints
+router
+    .route('/users')
+    .get(protected, (req, res) => {
```

Comment:
Solid! I like the use of protected here. Maybe this method can be use for the /api/restricted protected routes too. Since anything under them would be considered a protected route. Not sure how to 100% implement it this way yet, but its a good idea the way you are doing it for /users/